### PR TITLE
Read meta

### PR DESCRIPTION
**Pull request by chStaiger**

### CODE REVIEW

File: data/metadata_test/_validation_schema.toml [hunk 1]
```diff
@@ -97,7 +98,7 @@ type = "string"
 
 [[environment.columns]]
 name = "host_id"
-type = "string"
```
Comment (qubixes) [inline data/metadata_test/_validation_schema.toml:100]: Are you sure we need to specify the size of the integer? Perhaps int should be enough?

File: src/check_data.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from pathlib import Path
+
+
+def read_toml(path: Path) -> dict:
+    return toml.load(path)
+
+def read_csv(path: Path) -> pd.DataFrame:
+    return pd.read_csv(Path)
+
+def flatten_list_of_dicts(key: str, val: str, data: list) -> dict:
```
Comment (qubixes) [inline src/check_data.py:12]: Perhaps we should just adjust the input file, instead of having to convert it?

File: src/check_data.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    return undefined
+
+def duplicates(column_name: str, df: pd.DataFrame) -> pd.DataFrame:
+    return df[df[[column_name]].duplicated()][column_name]
+
+
+
+# read data and validation metadata
+df = pd.read_csv("data/dataE.csv")
```
Comment (qubixes) [inline src/check_data.py:70]: The validation might be imported at some point, so testing for `__name__ == "__main__"` is probably wise.

**Review by qubixes — CHANGES_REQUESTED** [review]
Thanks, it's a nice start. Are you planning to merge this soon or still waiting for more changes?

File: data/metadata_test/_validation_schema.toml (hunk 1 above)
Comment (chStaiger) [reply data/metadata_test/_validation_schema.toml:100]: That is how it comes out of pandas ... I do not like it that way. For now I will leave it in and we can refine from here.

File: src/check_data.py (hunk 3 above)
Comment (chStaiger) [reply src/check_data.py:103]: Yes. I agree. Something for the next PR?

Comment (qubixes) [reply src/check_data.py:103]: Sure

**Review by qubixes — APPROVED** [review]
Approved! We can modify things in other PRs, I think this is good enough for now!

File: src/check_data.py (hunk 2 above)
Comment (chStaiger) [reply src/check_data.py:12]: #2

**Comment by lagerweijgr** [comment]
With some delay, i approve

**Comment by chStaiger** [comment]
Replaces by #6

**Pull request closed without merging**